testindex/validators.py

```diff
@@ -121,6 +121,10 @@ def run(self):
                 else:
                     id_list.append(entry["id"])
 
+            # Check for pre-build script
+            if "prebuild-script" not in entry or ("prebuild-script" in entry and entry["prebuild-script"] is None):
+                self._summary_collector.add_warning("No pre-build script specified.")
```

Comment:
1. This will add warning for all the projects in index, which doesn't specify `prebuild-script`.
2. If checking for format of entry having `pre-build-script`, `add_error`should be used for `None` value of `prebuild-script` key.
3. Make it simple.

Reply:
@navidshaikh shaikh true. It will add a warning. But warning does not fail the validation.

 It should infact be a failure, but i am making it a warning for now, so it does not fail validation. We will first need update all index entries to include the new field, before i turn it into a failure.

This goes along the lines of @kbsingh ,wanting all entries in the index to be explicitly specified, even if it is None.

This brings us to the second point of what if a user does not have a prebuild-script. Then he should say None, and that goes out as a warning that it is not being used. 

The idea being index validation warning is its ok to have them, so long as there is valid justification. But people should be able to make that informed decision, before merging the PR.